File: Leetcode/Two Sum/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap();
        int complement;
        int size = nums.length;
        int[] ans = new int[2];
        for (int i = 0; i < size; ++i) {
            complement = target - nums[i];
            if (map.containsKey(complement)) {
                ans[0] = i;
                ans[1] = map.get(complement);
                return ans;
            } else {
                map.put(nums[i], i);
            }
        }
        return ans;
    }
}

File: Leetcode/Two Sum/Two Sum.py
"""
A python program to implement Two Sum
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

"""

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        container = {}
        
        for i, num in enumerate(nums):
            if target - num in container:
                return [container[target - num], i]
            container[num] = i
        return
        
"""
Sample input/output :
Example 1:
Input: nums = [2,7,11,15],
target = 9
Output: [0,1]

Example 2:
Input: nums = [3,2,4],
target = 6
Output: [1,2]

"""
